node/src/components/consensus/highway_core/validators.rs

```diff
@@ -142,6 +142,25 @@ impl<VID: Ord + Hash + fmt::Debug> fmt::Display for Validators<VID> {
 #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize, AsRef, From, Hash)]
 pub(crate) struct ValidatorMap<T>(Vec<T>);
 
+impl<T> fmt::Display for ValidatorMap<Option<T>>
+where
+    T: fmt::Display,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let view = self
+            .0
+            .iter()
+            .map(|maybe_el| match maybe_el {
+                None => "N".to_string(),
+                Some(el) => format!("{}", el),
+            })
+            .collect::<Vec<_>>()
+            .join(", ");
+        write!(f, "OldestSeen({})", view)?;
```

Comment:
Since this is `ValidatorMap`'s `Display` impl and not necessarily only for this purpose, maybe:
```suggestion
        write!(f, "validator map: {}", view)?;
```
Or: Create a wrapper type in the `synchronizer` module, with this more specific `Display` impl?

Reply:
Ah, good point.